src/transbank.c

```diff
@@ -66,6 +66,79 @@ int open_port(char* portName, int baudrate){
   return retval;
 }
 
+char* substring(char* string, ParamInfo info){
+  char* ret = malloc((sizeof(char) * info.length) +1);
+  memset(ret, '\0', sizeof(ret));
+  char cToStr[2];
+  cToStr[1] = '\0';
+  for (int i = info.index; i < (info.index + info.length); i++){
```

Comment:
`info.index + info.length` What is this sum for?  Can you move it to a var?

Reply:
This function extract a substring from the original message, in order to parse it inside a structure.

`info.index + info.length` is the last position in the array which has to be stracted. I'll move it into a variable to make this clearer.